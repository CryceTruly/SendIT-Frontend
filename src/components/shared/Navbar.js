import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import Nav from "../../views/Navbar";
import { logout } from "../../actions/authActions";

export class Navigationbar extends Component {
  
    onLogoutClick=() => {
      this.props.logout(this.props);
    }

    render() {
      const { isAuthenticated, user } = this.props.auth;
      return (
        <Nav isAuthenticated={isAuthenticated} user={user} onLogoutClick={this.onLogoutClick} />
      );
    }
}

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { logout })(Navigationbar);
