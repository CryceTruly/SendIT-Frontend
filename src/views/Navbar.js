import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Nav = (props) => {
  const { isAuthenticated, user,onLogoutClick } = props;

  const authLinks = (
    <ul className="navbar-nav ml-auto">
      <form className="form-inline my-2 my-lg-0">
        <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" />
      </form>

      <li className="nav-item">
        <Link className="nav-link" to="/">  <strong>{user ? `Welcome ${user.username}` : ""}</strong> </Link>
      </li>
      <li className="nav-item">
        {user && user.is_admin ? (<Link className="nav-link" to="/new">All Activity</Link>) : (<Link className="nav-link" to="/new">Create Order</Link>
        )}
      </li>
      <li className="nav-item" onClick={onLogoutClick}>
        <Link className="nav-link" to="">Logout</Link>
      </li>
    </ul>
  );

  const guestLinks = (
    <ul className="navbar-nav ml-auto">
      <li className="nav-item">
        <Link className="nav-link" to="/login">Login </Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to="/register"> Register</Link>
      </li>
    </ul>
  );
  return (

    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container">
        <span className="navbar-brand"><i class="fa fa-paper-plane" aria-hidden="true"></i>SendIT</span>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          {isAuthenticated ? authLinks : guestLinks}
        </div>
      </div>
    </nav>
  );
};

export default Nav;
