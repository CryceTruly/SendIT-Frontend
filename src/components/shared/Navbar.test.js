import React from "react";
import {
  shallow, mount,
} from "enzyme";
import Provider from "react-redux";
import { Navigationbar, mapStateToProps } from "./Navbar";
import store from "../../Store";

describe("Nav TestSuite", () => {
  const props = {
    auth: {
      isAuthenticated: true,
      user: {
        email: "crycetruly@gmail.com",
        username: "fredgreg",
      },
    },
    logout: jest.fn(),
  };

  it("Should not regress", () => {
    const wrapper = shallow(<Navigationbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle user logout", () => {
    const wrapper = shallow(<Navigationbar {...props} />);
    wrapper.instance().onLogoutClick();
  });

  test("should map state to props", () => {
    const auth = {
      user_id: localStorage.getItem("user_id"),
      isAuthenticated: null,
      isLoading: false,
      user: null,
    };
    const appState = { auth };
    const componentState = mapStateToProps(appState);
    expect(componentState).toEqual(appState);
  });
});
