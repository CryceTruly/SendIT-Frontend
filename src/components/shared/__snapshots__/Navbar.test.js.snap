// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav TestSuite Should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigationbar
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "email": "crycetruly@gmail.com",
          "username": "fredgreg",
        },
      }
    }
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "isAuthenticated": true,
      "onLogoutClick": [Function],
      "user": Object {
        "email": "crycetruly@gmail.com",
        "username": "fredgreg",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "isAuthenticated": true,
        "onLogoutClick": [Function],
        "user": Object {
          "email": "crycetruly@gmail.com",
          "username": "fredgreg",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
